### DISCUSSION TITLE
D-Pad Won't Function

### DISCUSSION THREAD

**Discussion by MarcelineVampQueen** [nefarius/DsHidMini#108] [Q&A] [answered]
Trying to play Dead Space, and the d-pad, which is used to swap weapons, doesn't seem to work in any form. Not only that -- pressing right on the d-pad crashes the game. Dunno why. Will keep testing and trying the different controller modes to see if that helps, though.

**Comment by MarcelineVampQueen** [comment]
(Pressed enter too early when making the issue, hence the edits. My bad.)

**Comment by MarcelineVampQueen** [comment]
And, if you're curious, it does NOT seem to be a fault of the game. I plugged in a different PS3 controller before installing DSHID, a knock-off, and it worked just fine, (still does after the install) but the X button on that one was broken (physically -- not a software issue). I'm going to plug that one back in and see what windows says its dpad inputs are registered as, though -- will report back with the results.

**Comment by MarcelineVampQueen** [comment]
...So the knockoff doesn't classify itself as a controller, though it has "HID-Compliant game controller" listed as a function under the hardware tab. Interesting. Also more evidence to it being a driver issue, as it doesn't use this driver due to it "not" being a controller.
![image](https://user-images.githubusercontent.com/87347441/125377413-8e59b800-e352-11eb-81c8-4713ed16cd46.png)
Unfortunately, this means that I can't measure its inputs due to windows not recognizing it as a controller.

**Comment by MarcelineVampQueen** [comment]
Hmmm. Further testing shows that Windows measures the inputs fine with the SXS mode -- no slip-ups there. Maybe it's because the D-Pad isn't registering as a group of buttons? As in, when I press things on the D-Pad, its direction shows up on the "point of view hat", but none of the 24 (are there really THAT many?) button indicators light up. 
![image](https://user-images.githubusercontent.com/87347441/125379063-746da480-e355-11eb-8b1e-84eb8aed23ac.png)
...but then why would right d-pad do anything, much less crash the game?
Not a driver issue -- at least, not entirely -- then.

But if not that, WHAT? The knock-off worked fine, and the only difference IS the driver's presence, isn't it?
Is it a windows-interacting-with-the-game issue? A game-interacting-with-the-controller issue?

Blegh. I've spent half my day trying to figure this out.
Y'know what, I'm just gonna play on mouse and keyboard, this ain't worth the effort -- even if it's a thousand times better on controller.
(but if you do figure out anything about this though, please let me know.)

**Comment by nefarius** [comment, marked as answer]
You could've saved yourself **a lot** of time and frustration if you've simply have read the documentation 😊 for games expecting something like an Xbox controller **use DS4Windows mode** to proxy your DS3 through to an emulated x360 pad. The links on how to do are everywhere. Good luck!

**Reply by MarcelineVampQueen** [reply]
Oh, okay. Thanks!
